source/elements/oneMKL/source/domains/spblas/data_types/create_coo_matrix.rst

```diff
@@ -0,0 +1,241 @@
+      initialized ``matrix_handle_t`` object that can be filled and used to
+      perform sparse BLAS operations.
+
+.. container:: section
+
+   .. rubric:: Throws
+
+   This function shall throw the following exceptions if the associated
+   condition is detected. An implementation may throw additional
+   implementation-specific exception(s) in case of error conditions not covered
+   here.
+
+   | :ref:`oneapi::mkl::device_bad_alloc<onemkl_exception_device_bad_alloc>`
+   | :ref:`oneapi::mkl::host_bad_alloc<onemkl_exception_host_bad_alloc>`
+   | :ref:`oneapi::mkl::invalid_argument<onemkl_exception_invalid_argument>`
+   | :ref:`oneapi::mkl::unimplemented<onemkl_exception_unimplemented>`
+   | :ref:`oneapi::mkl::unsupported_device<onemkl_exception_unsupported_device>`
+
+.. _onemkl_sparse_create_coo_matrix_usm:
+
+create_coo_matrix (USM version)
+-------------------------------
+
+.. rubric:: Syntax
+
+.. code-block:: cpp
+
+   namespace oneapi::mkl::sparse {
+
+       template <typename dataType, typename indexType>
+       void create_coo_matrix (sycl::queue                          &queue,
+                               oneapi::mkl::sparse::matrix_handle_t *p_smhandle,
+                               std::int64_t                         num_rows,
+                               std::int64_t                         num_cols,
+                               std::int64_t                         nnz,
+                               index_base                           index,
+                               indexType                            *row_ind,
+                               indexType                            *col_ind,
+                               dataType                             *val);
+
+   }
+
+.. container:: section
+
+   .. rubric:: Template parameters
+
+   dataType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+   indexType
+      See :ref:`supported template types<onemkl_sparse_supported_types>`.
+
+.. container:: section
+
+   .. rubric:: Input parameters
+
+   queue
+      The SYCL command queue which will be used for SYCL kernels execution.
+
+   p_smhandle
+      The address of the ``p_smhandle`` object to be initialized. Must only be
+      called on an uninitialized ``matrix_handle_t`` object.
+
+   num_rows
+      Number of rows of the provided data ``val``. Must be at least 0.
+
+   num_cols
+      Number of columns of the provided data ``val``. Must be at least 0.
+
+   nnz
+      Number of non-zero entries in the matrix (which may include explicit
+      zeros). Must be at least 0.
+
+   index
+      Indicates how input arrays are indexed. The possible options are described
+      in :ref:`onemkl_enum_index_base` enum class.
+
+   row_ind
+      USM pointer of length ``nnz`` containing the row indices in
+      ``index``-based numbering. Refer to :ref:`onemkl_sparse_coo` format for
+      detailed description of ``row_ind``. The data must be accessible on the
+      device.
```

Comment:
I wonder if we should specify somewhere that using `row_ind`/`col_ind`/`val` USM pointers that have been allocated with length `<nnz` is undefined behavior since there's no way for us to confirm that at run-time.

Reply:
Addressed in https://github.com/oneapi-src/oneAPI-spec/pull/522/commits/b7037b5dd74b9055e69f2682ae55a2b01311a001